pyat/at/lattice/axisdef.py

```diff
@@ -0,0 +1,164 @@
+"""Helper functions for axis and plane descriptions"""
+from __future__ import annotations
+from typing import Optional, Union
+# For sys.version_info.minor < 9:
+from typing import Tuple
+
+AxisCode = Union[str, int, slice, None, type(Ellipsis)]
+AxisDef = Union[AxisCode, Tuple[AxisCode, AxisCode]]
+
+_axis_def = dict(
+    x=dict(index=0, label="x", unit=" [m]"),
+    px=dict(index=1, label=r"$p_x$", unit=" [rad]"),
+    y=dict(index=2, label="y", unit=" [m]"),
+    py=dict(index=3, label=r"$p_y$", unit=" [rad]"),
+    dp=dict(index=4, label=r"$\delta$", unit=""),
+    ct=dict(index=5, label=r"$\beta c \tau$", unit=" [m]"),
+)
+for xk, xv in [it for it in _axis_def.items()]:
+    xv['code'] = xk
+    _axis_def[xv['index']] = xv
+    _axis_def[xk.upper()] = xv
+_axis_def['xp'] = _axis_def['px']
+_axis_def['yp'] = _axis_def['py']
+_axis_def['delta'] = _axis_def['dp']
+_axis_def['s'] = _axis_def['ct']
+_axis_def[None] = dict(index=slice(None), label="", unit="", code=":")
+_axis_def[Ellipsis] = dict(index=Ellipsis, label="", unit="", code="...")
+
+_plane_def = dict(
+    h=dict(index=0, label="x", unit=" [m]"),
+    v=dict(index=1, label="y", unit=" [m]"),
+    l=dict(index=2, label="z", unit="")
+)
+for xk, xv in [it for it in _plane_def.items()]:
+    xv['code'] = xk
+    _plane_def[xv['index']] = xv
+    _plane_def[xk.upper()] = xv
+_plane_def['h'] = _plane_def['x']
+_plane_def['v'] = _plane_def['y']
+_plane_def[None] = dict(index=slice(None), label="", unit="", code=":")
+_plane_def[Ellipsis] = dict(index=Ellipsis, label="", unit="", code="...")
+
+
+def _descr(dd: dict, arg: AxisDef, key: Optional[str] = None):
+    if isinstance(arg, tuple):
+        return tuple(_descr(dd, a, key=key) for a in arg)
+    else:
+        try:
+            descr = dd[arg]
+        except (TypeError, KeyError):
+            descr = dict(index=arg, code=arg, label="", unit="")
+        if key is None:
+            return descr
+        else:
+            return descr[key]
+
+
+def axis_(axis: AxisDef, key: Optional[str] = None):
+    r"""Return axis descriptions
+
+    Parameters:
+        axis:           axis code or tuple of axis codes selecting axes in the
+          standard AT coordinate system. codes for the 6 axes are:
+
+          0, 'x', 'X'
+
+          1, 'px', PX', 'xp'
+
+          2, 'z', 'Z'
```

Comment:
I did not see this.... `z` notation for the vertical axis is really not used by many anymore and for very good reasons.... in case you insist in keeping it could you at least add `y` and `Y` to the list?

Reply:
Typo: should be y and Y